python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/io_manager.py

```diff
@@ -326,10 +351,38 @@ def create_io_manager(self, context) -> Generator:
 
 
 class BigQueryClient(DbClient):
-    @staticmethod
-    def delete_table_slice(context: OutputContext, table_slice: TableSlice, connection) -> None:
+    def __init__(self, write_mode: Optional[Union[BigQueryWriteMode, str]] = BigQueryWriteMode.TRUNCATE):
+        # Coerce string inputs (from raw config) to the enum, fallback to TRUNCATE on invalid values
+        if write_mode is None:
+            write_mode = BigQueryWriteMode.TRUNCATE
+            
+        if isinstance(write_mode, str):
+            try:
+                write_mode = BigQueryWriteMode(write_mode)
+            except ValueError:
+                write_mode = BigQueryWriteMode.TRUNCATE
```

Comment:
If an invalid value was passed, we should error instead of defaulting to `TRUNCATE`. We don't want a type (`"RPLACE"`) mapping to `TRUNCATE`